cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala

```diff
@@ -441,16 +836,287 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
+    profileQuery(
+      title = "Let Semi Apply",
+      text =
+        """The `LetSemiApply` operator tests for the existence of a pattern predicate.
+          |When a query contains multiple pattern predicates separated with `OR`, `LetSemiApply` will be used to evaluate the first of these.
+          |It will record the result of evaluating the predicate but will leave any filtering to another operator.
+          |In the example, `LetSemiApply` will be used to check for the existence of the `FRIENDS_WITH`
+          |relationship from each person.""".stripMargin,
+      queryText =
+        """MATCH (other:Person)
+          |WHERE (other)-[:FRIENDS_WITH]->(:Person) OR (other)-[:WORKS_IN]->(:Location)
+          |RETURN other.name""".stripMargin,
+      assertions = (p) => assertThat(p.executionPlanDescription().toString, containsString("LetSemiApply"))
+    )
+  }
+
+  @Test def letAntiSemiApply() {
+    profileQuery(
+      title = "Let Anti Semi Apply",
+      text =
+        """The `LetAntiSemiApply` operator tests for the absence of a pattern.
+          |When a query contains multiple negated pattern predicates -- i.e. predicates separated with `OR`, where at least one predicate contains `NOT` -- `LetAntiSemiApply` will be used to evaluate the first of these.
+          |It will record the result of evaluating the predicate but will leave any filtering to another operator.
+          |In the example, `LetAntiSemiApply` will be used to check for the absence of
+          |the `FRIENDS_WITH` relationship from each person.""".stripMargin,
+      queryText =
+        """MATCH (other:Person)
+          |WHERE NOT ((other)-[:FRIENDS_WITH]->(:Person)) OR (other)-[:WORKS_IN]->(:Location)
+          |RETURN other.name""".stripMargin,
+      assertions = (p) => assertThat(p.executionPlanDescription().toString, containsString("LetAntiSemiApply"))
+    )
+  }
+
+  @Test def selectOrSemiApply() {
+    profileQuery(
+      title = "Select Or Semi Apply",
+      text =
+        """The `SelectOrSemiApply` operator tests for the existence of a pattern predicate and evaluates a predicate.
+          |This operator allows for the mixing of normal predicates and pattern predicates
+          |that check for the existence of a pattern.
+          |First, the normal expression predicate is evaluated, and, only if it returns `false`, is the costly pattern predicate evaluated.""".stripMargin,
+      queryText =
+        """MATCH (other:Person)
+          |WHERE other.age > 25 OR (other)-[:FRIENDS_WITH]->(:Person)
+          |RETURN other.name""".stripMargin,
+      assertions = (p) => assertThat(p.executionPlanDescription().toString, containsString("SelectOrSemiApply"))
+    )
+  }
+
+  @Test def selectOrAntiSemiApply() {
+    profileQuery(
+      title = "Select Or Anti Semi Apply",
+      text =
+        """The `SelectOrAntiSemiApply` operator tests for the absence of a pattern predicate and evaluates a predicate.""".stripMargin,
```

Comment:
The`SelectOrAntiSemiApply` operator is used to evaluate `OR` between a predicate and a negative pattern predicate. If the predicate returns `true`, the pattern is not tested. If the predicate returns `false` or `null`, the operator will test the pattern instead.

Reply:
👍